src/Pos/Block/Network/Logic.hs

```diff
@@ -64,6 +68,27 @@ import           Pos.Util                   (inAssertMode, _neHead, _neLast)
 import           Pos.Util.Chrono            (NE, NewestFirst (..), OldestFirst (..))
 import           Pos.WorkMode               (WorkMode)
 
+
+----------------------------------------------------------------------------
+-- Exceptions
+----------------------------------------------------------------------------
+
+data BlockNetLogicException
+    = VerifyBlocksException Text
+      -- ^ Failed to verify blocks coming from node.
+    | DialogUnexpected Text
+      -- ^ Node's response in any network/block related logic was
+      -- unexpected.
+    deriving (Show)
+
+instance B.Buildable BlockNetLogicException where
+    build e = bprint ("BlockNetLogicException: "%shown) e
```

Comment:
`Buildable` instance is required to derive from `CardanoException`, but i didn't want to provide anything too complex.